hw/ip/otbn/dv/rig/rig/model.py

```diff
@@ -779,6 +779,94 @@ def update_for_bnmovr(self, prog_insn: ProgInsn) -> None:
+        assert len(insn.operands) == len(op_vals)
+
+        exp_shape = (isinstance(insn.operands[0].op_type, RegOperandType) and
+                     insn.operands[0].op_type.reg_type == 'wdr' and
+                     insn.operands[0].op_type.is_dest() and
+                     isinstance(insn.operands[1].op_type, RegOperandType) and
+                     insn.operands[1].op_type.reg_type == 'wdr' and
+                     not insn.operands[1].op_type.is_dest() and
+                     isinstance(insn.operands[2].op_type, RegOperandType) and
+                     insn.operands[2].op_type.reg_type == 'wdr' and
+                     not insn.operands[2].op_type.is_dest() and
+                     isinstance(insn.operands[4].op_type, ImmOperandType))
+        if not exp_shape:
+            raise RuntimeError('BN.XOR instruction read from insns.yml is '
+                               'not the shape expected by '
+                               'Model.update_for_bnxor.')
+
+        wrs1_val = self.get_reg('wdr', op_vals[1])
+        wrs2_val = self.get_reg('wdr', op_vals[2])
+
+        result = None
+
+        if (op_vals[1] == op_vals[2]) and (op_vals[4] == 0):
+            result = 0
+        elif wrs1_val is None or wrs2_val is None:
+            pass
+        else:
+            if op_vals[3]:
+                src2_val = wrs2_val >> op_vals[4]
+                result = (wrs1_val ^ src2_val)
+            else:
+                src2_val = wrs2_val << op_vals[4]
+                result = (wrs1_val ^ src2_val)
+
+        self._generic_update_for_insn(prog_insn)
+
+        self.write_reg('wdr', op_vals[0], result, True)
+
+    def update_for_bnnot(self, prog_insn: ProgInsn) -> None:
+        '''Update model state after an BN.NOT
+
+        If the source register happens to have a known value, we can do the
+        addition and store the known result.
+
+        '''
+        insn = prog_insn.insn
+        op_vals = prog_insn.operands
+        assert insn.mnemonic == 'bn.not'
+        assert len(insn.operands) == len(op_vals)
+
+        exp_shape = (isinstance(insn.operands[0].op_type, RegOperandType) and
+                     insn.operands[0].op_type.reg_type == 'wdr' and
+                     insn.operands[0].op_type.is_dest() and
+                     isinstance(insn.operands[1].op_type, RegOperandType) and
+                     insn.operands[1].op_type.reg_type == 'wdr' and
+                     not insn.operands[1].op_type.is_dest() and
+                     isinstance(insn.operands[2].op_type, EnumOperandType) and
+                     isinstance(insn.operands[3].op_type, ImmOperandType))
+        if not exp_shape:
+            raise RuntimeError('BN.NOT instruction read from insns.yml is '
+                               'not the shape expected by '
+                               'Model.update_for_bnnot.')
+
+        wrs_val = self.get_reg('wdr', op_vals[1])
+
+        if wrs_val is None:
+            result = None
+        else:
```

Comment:
This is fine, but I'd consider writing it like this:
```python3
    result = None
    if wrs_val is not None:
       ...
```
which has less confusing scoping (I'm not a fan of Python's scoping rules!)